app/handlers/base.py

```diff
@@ -152,6 +153,33 @@ def bulk_verify(mode, collection, accessor):
 
 
 class BaseHandler(PSABaseHandler):
+    @contextmanager
+    def Session(self, use_auto_verify=True):
+        """
+        Generate a scoped session that also has knowledge
+        of the current user, and that can automatically
+        verify and commit the changes to it when going out of context.
+        The current user is taken from the handler's current_user.
+        This is a shortcut method to models.Session
+        that saves the need to manually input the user object.
+
+        Parameters
+        ----------
+        use_auto_verify: boolean
+            if True (default), will call verify and commit
+            functions of the session before exiting the context.
```

Comment:
```suggestion
            objects in the session before exiting the context.
```

Reply:
I think the behavior of this should be to verify *upon* commit, i.e. to override the commit of the returned Session.

Reply:
Not sure what you mean...?

Reply:
You mean that every time `commit()` is called it will also call `verify()`? That means we will also need to have a switch controlling whether we want to automatically commit at end of session, on top of the one that makes it automatically verify before each commit.

Reply:
I don't think we want to modify the behavior of existing sessions, other than to add verification. Sessions do not automatically commit, so it doesn't make sense to automatically verify.  Commit should trigger a verify for our special session.

Reply:
I think it could be nice to produce a session that also commits and verifies when exiting. That's an optional thing, so we can have it default to False, but still allow the user to start a session, in context, that would (verify and) commit at exit.

Reply:
I think a nice-to-have does not outweigh the inconsistency this will introduce with SQLAlchemy.  The risk of introducing programming errors because users are looking at the SQLAlchemy docs is too big, especially since it is done silently.

And we don't really need it, do we? It's mainly a convenience for not typing `session.commit()` when you're ready to write to the DB?

Reply:
I think having `session.commit()` verify, on the other hand, is a perfectly reasonable addition.

Reply:
I guess that's true. I was just thinking that while we are adding stuff at the end of the context, might as well do that also. But I will remove it.